packages/valory/skills/market_creation_manager_abci/behaviours.py

```diff
@@ -2443,14 +2285,26 @@ def _get_balance(self, account: str) -> Generator[None, None, Optional[int]]:
     def get_payload(self) -> Generator[None, None, str]:
         """Get the transaction payload"""
         # get the questions to that need to be answered
-        questions = yield from self._get_unanswered_questions()
-        if questions is None:
+        unanswered_questions = yield from self._get_unanswered_questions()
+
+        if unanswered_questions is None:
             self.context.logger.info("Couldn't get the questions")
-            return CloseMarketsRound.ERROR_PAYLOAD
+            return GetPendingQuestionsRound.ERROR_PAYLOAD
+
+        filtered_questions = [
+            question
+            for question in unanswered_questions
+            if question["question"]["id"].lower()
+            not in self.shared_state.questions_requested_mech
+        ]
+        random.seed(self.last_synced_timestamp)
+        num_questions = min(len(filtered_questions), self.params.multisend_batch_size)
```

Comment:
@jmoreira-valory 

The `params` property does not specify the `multisend_batch_size`.